redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -426,42 +435,74 @@ object Input {
     _10: Input[J],
     _11: Input[K]
   ) extends Input[(A, B, C, D, E, F, G, H, I, J, K)] {
-    def encode(data: (A, B, C, D, E, F, G, H, I, J, K)): Chunk[RespValue.BulkString] =
+    def encode(data: (A, B, C, D, E, F, G, H, I, J, K))(implicit codec: Codec): Chunk[RespValue.BulkString] =
       _1.encode(data._1) ++ _2.encode(data._2) ++ _3.encode(data._3) ++ _4.encode(data._4) ++
         _5.encode(data._5) ++ _6.encode(data._6) ++ _7.encode(data._7) ++ _8.encode(data._8) ++
         _9.encode(data._9) ++ _10.encode(data._10) ++ _11.encode(data._11)
   }
 
   case object UpdateInput extends Input[Update] {
-    def encode(data: Update): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
+    def encode(data: Update)(implicit codec: Codec): Chunk[RespValue.BulkString] = Chunk.single(encodeString(data.stringify))
   }
 
   final case class Varargs[-A](input: Input[A]) extends Input[Iterable[A]] {
-    def encode(data: Iterable[A]): Chunk[RespValue.BulkString] =
+    def encode(data: Iterable[A])(implicit codec: Codec): Chunk[RespValue.BulkString] =
       data.foldLeft(Chunk.empty: Chunk[RespValue.BulkString])((acc, a) => acc ++ input.encode(a))
   }
 
+  case object EvalInput extends Input[(String, Chunk[Chunk[Byte]], Chunk[Chunk[Byte]])] {
+    def encode(data: (String, Chunk[Chunk[Byte]], Chunk[Chunk[Byte]]))(implicit codec: Codec): Chunk[RespValue.BulkString] = {
+      val (lua, keys, args) = data
+      val encodedScript     = Chunk(encodeString(lua), encodeString(keys.size.toString))
+      val encodedKeys       = keys.foldLeft[Chunk[RespValue.BulkString]](Chunk.empty)((cur, next) => cur :+ encodeBytes(next))
+      val encodedArgs       = args.foldLeft[Chunk[RespValue.BulkString]](Chunk.empty)((cur, next) => cur :+ encodeBytes(next))
+      encodedScript ++ encodedKeys ++ encodedArgs
+    }
+  }
+
```

Comment:
Scripting part should stay in its own pull request, please don't introduce it here.